client/core/core.go

```diff
@@ -1899,6 +1899,33 @@ func (c *Core) ReconfigureWallet(appPW, newWalletPW []byte, assetID uint32, cfg
 		return err
 	}
 
+	// If there are active trades, make sure they can be settled by the
+	// keys held within the new wallet.
+	sameWallet := func() error {
+		for _, dc := range c.dexConnections() {
+			for _, trade := range dc.trackedTrades() {
+				if trade.isActive() {
+					addr := trade.Trade().Address
+					owns, err := wallet.OwnsAddress(addr)
```

Comment:
This is only the right address for the "from asset" (quote asset + buy order, or base asset + sell order). For the other asset, you could in theory check the refund address, but `Core` doesn't store that and it's not part of the `*asset.Contract` right now. Hmm. Not sure what to do.

I have to take blame for this situation, since I declined to store the refund address in the `OrderMetadata` and even removed access to the refund address when `asset.Contract` was changed from an interface to a concrete type.

Reply:
I think it's now checking refund addresses should it find any. Do I have my from/to asset terminology backwards?